src/components/Home.tsx

```diff
@@ -22,15 +23,19 @@ export type MoviesType = {
 const Home = () => {
   const [movies, setMovies] = useState<[]>([]);
   const [history, setHistory] = useState<MoviesType[]>();
+  const [popularMovies, setPopularMovies] = useState<Movie[]>([]);
 
   useEffect(() => {
-    api.getPopularQueryList().then(res => setMovies(res.slice(0, 10)));
+    api.getPopularQueryList().then(res => {
+      setMovies(res.slice(0, 10));
+      setPopularMovies(res);
+    });
     api.getDataByIds(api.getHistoryIdList()).then((res) => { setHistory(res) });
   }, []);
 
   return (
     <>
-      <MovieBanner />
+     {popularMovies.length && <MovieBanner popularMovies={popularMovies}/>}
```

Comment:
I dont think we need to have this check here... Remove this check please

Reply:
I thought if we use here popularMovies.length we will be able to not pass popularMovies as undefined to a MovieBanner